pallets/ajuna-awesome-avatars/src/types/avatar/versions/v2/mod.rs

```diff
@@ -209,69 +208,96 @@ impl<T: Config> ForgerV2<T> {
 		let leader_item_type = AvatarUtils::read_item_type::<ItemType>(leader);
 
 		match leader_item_type {
-			ItemType::Pet => {
-				let leader_sub_type = AvatarUtils::read_sub_type::<PetItemType>(leader);
-
-				match leader_sub_type {
-					PetItemType::Pet => {
-						if sacrifices.iter().all(|sacrifice| {
-							let equippable_item = AvatarUtils::read_attribute_as(
+			ItemType::Pet => match AvatarUtils::read_sub_type::<PetItemType>(leader) {
+				PetItemType::Pet => {
+					if sacrifices.iter().all(|sacrifice| {
+						let equippable_item = AvatarUtils::read_attribute_as(
+							sacrifice,
+							&AvatarAttributes::ItemSubType,
+						);
+						AvatarUtils::has_item_type(sacrifice, ItemType::Equippable) &&
+							AvatarUtils::same_rarity_tier(leader, sacrifice) &&
+							AvatarUtils::same_class_type2(leader, sacrifice) &&
+							(equippable_item == EquippableItemType::ArmorBase ||
+								EquippableItemType::is_weapon(equippable_item))
+					}) {
+						ForgeType::Equip
+					} else if sacrifices.iter().all(|sacrifice| {
+						AvatarUtils::same_item_type(sacrifice, leader) &&
+							AvatarUtils::same_item_sub_type(sacrifice, leader) &&
+							AvatarUtils::same_rarity_tier(sacrifice, leader)
+					}) {
+						ForgeType::Mate
+					} else if sacrifices.iter().all(|sacrifice| {
+						AvatarUtils::same_item_type(sacrifice, leader) &&
+							AvatarUtils::has_item_sub_type(sacrifice, PetItemType::Egg) &&
+							AvatarUtils::read_rarity(sacrifice) < RarityTier::Legendary
+					}) {
+						ForgeType::Feed
+					} else {
+						ForgeType::None
+					}
+				},
+				PetItemType::PetPart => {
+					if AvatarUtils::has_attribute_set_with_values(
+						leader,
+						&[
+							(AvatarAttributes::ClassType1, HexType::X0.as_byte()),
```

Comment:
recurring checks like this need to encapsulated into some thing like IsGenericPetPart() which checks for ClassType1 == X0 and ClassType2 == X0